pkg/queue/manager.go

```diff
@@ -553,15 +553,42 @@ func (m *Manager) reportPendingWorkloads(cqName string, cq ClusterQueue) {
 	metrics.ReportPendingWorkloads(cqName, active, inadmissible)
 }
 
-func (m *Manager) GetClusterQueues() map[string]ClusterQueue {
+func (m *Manager) GetClusterQueueNames() []string {
 	m.RLock()
 	defer m.RUnlock()
-	return m.clusterQueues
+	clusterQueueNames := make([]string, 0, len(m.clusterQueues))
+	for k := range m.clusterQueues {
+		clusterQueueNames = append(clusterQueueNames, k)
+	}
+	return clusterQueueNames
+}
+
+func (m *Manager) getClusterQueue(cqName string) ClusterQueue {
+	m.RLock()
+	defer m.RUnlock()
+	return m.clusterQueues[cqName]
 }
 
-func (m *Manager) SetSnapshot(cqName string, workloads []kueue.ClusterQueuePendingWorkload) {
+func (m *Manager) UpdateSnapshot(cqName string, maxCount int32) {
 	m.snapshotsMutex.Lock()
```

Comment:
This mutex is only necessary for `m.snapshots[cqName] = workloads`. I suggest to extract this to `setSnapshot(cqName, workloads)`, and take the lock there.

Reply:
added here https://github.com/kubernetes-sigs/kueue/pull/1069/commits/20448ee40ed7f91bb885cdca263f5e10fe1bbe5b